### PR TITLE
Add support for certificate uploads

### PR DESCRIPTION
**Pull request by jonandernovella**
- Issuing and updating endpoints can now upload the pdfs to S3.
- boto3 library used for interacting with S3
- minio dev server is used

### CODE REVIEW

File: app/herdbook.py [hunk 1]
```diff
@@ -513,14 +523,22 @@ def issue_certificate(i_number):
     if certificate_exists:
         return jsonify({"response": "Certificate already exists"}), 400
 
-    ind.update(**request.form)
+    ## TODO: Make sure this is unique
+    cert_number = str(random.randint(50001, 99999))
```
Comment (pontus) [inline app/herdbook.py:527]: There is a todo but these would normally be a series, so we should keep track and issue the next.

File: app/utils/s3.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        secret_key,
+        access_key,
+        verify,
+        cert,
+        private_key,
+    ):
+        """
+        Constructor for initialising a S3 client.
+        """
+        assert isinstance(bucket, str)
```
Comment (pontus) [inline app/utils/s3.py:29]: I'm personally not overall fond of this kind - if something can fulfill the role it should be okay, and alterantively if one wants stronger typing, I prefer using `typing` and static tools to analyse.

File: app/utils/s3.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            endpoint_url=self.endpoint,
+            region_name=region,
+            use_ssl=self.endpoint.startswith("https"),
+            verify=verify,
+            aws_access_key_id=access_key,
+            aws_secret_access_key=secret_key,
+            config=config,
+        )
+
+        print(vars(self))
```
Comment (pontus) [inline app/utils/s3.py:65]: These are easy to forget, better to write a suitable log statement directly?

File: app/herdbook.py (hunk 1 above)
Comment (jonandernovella) [reply app/herdbook.py:527]: This will be fixed in a future PR.

File: app/utils/s3.py (hunk 2 above)
Comment (jonandernovella) [reply app/utils/s3.py:29]: I am not either. But it felt like a quick stopper for undesired values passed into the configuration. We could try out [Mypy](https://github.com/python/mypy) in the next iteration?

File: app/utils/s3.py (hunk 3 above)
Comment (jonandernovella) [reply app/utils/s3.py:65]: Yes. Good that you found this one. We can set it as a trace or debug line.

**Pull request merged**